models/fates/R/model2netcdf.FATES.R

```diff
@@ -10,581 +10,143 @@
+        nc_month_names <- names(nc_month$var)
         
+        ## Create time bounds to populate time_bounds variable iteratively
+        var_bound <- ncdf4::ncvar_get(nc_month, "time_bounds") # start,end day of month
         
-        ## FATES time is in multiple columns, create 'time'
-        mcdate <- ncdf4::ncvar_get(ncin, "mcdate")                  # current date (YYYYMMDD)
-        if (length(mcdate)==1) {
-          ## do we need to bother converting outputs where FATES provides only a single timepoint for a date?
-          ## usually happens when the model starts/finishes at the end/start of a new year
-          PEcAn.logger::logger.debug("*** Skipping conversion for output with only a single timepoint ***")
-          next
-        }
-        cal_dates <- as.Date(as.character(mcdate),format="%Y%m%d")  # in standard YYYY-MM-DD format
-        julian_dates <- lubridate::yday(cal_dates)                  # current year DOY values
-        day  <- ncdf4::ncvar_get(ncin, "mdcur")                     # current day (from base day)
-        sec  <- ncdf4::ncvar_get(ncin, "mscur")                     # current seconds of current day
-        nstep <- ncdf4::ncvar_get(ncin, "nstep")                    # model time step
-        time <- day + sec / 86400                                   # fractional time since base date (typically first day of full model simulation)
-        iter_per_day <- length(unique(sec))                         # how many outputs per day (e.g. 1, 24, 48)
-        timesteps <- utils::head(seq(0, 1, by = 1 / iter_per_day), -1)     # time of day fraction
-        current_year_tvals <- (julian_dates-1 + timesteps)          # fractional DOY of current year
-        nt <- length(time)                                          # output length
-        nc_time <- ncin$dim$time$vals                               # days since "start_date"
-
-        # !! Is this a useful/reasonable check? That is that our calculated time
-        # matches FATES internal time var.
-        if (length(time)!=length(nc_time)) {
-          PEcAn.logger::logger.severe("Time dimension mismatch in output, simulation error?")
-        }
-
-        ## Create time bounds to populate time_bounds variable
-        bounds <- array(data = NA, dim = c(length(time), 2))
-        bounds[, 1] <- time
-        bounds[, 2] <- bounds[, 1] + (1 / iter_per_day)
-        bounds <- round(bounds, 4)  # create time bounds for each timestep in t, t+1; t+1, t+2... format
-
-        #******************** Declare netCDF dimensions ********************#
-        nc_var  <- list()
-        sitelat <- ncdf4::ncvar_get(ncin,"lat")
-        sitelon <- ncdf4::ncvar_get(ncin,"lon")
-        ## time variable based on internal calc, nc$dim$time is the FATES output time
+        ## define dimensions 
         t <- ncdf4::ncdim_def(name = "time", units = paste0("days since ", init_year, "-01-01 00:00:00"),
```

Comment:
you seem to be using `init_year` without initializing it (previous version initializes it above with `init_year <- unique(years)[1]`)